src/analyses/recent.js

```diff
@@ -8,50 +8,29 @@
 
 import logger from "../logging";
 import { validateLimit } from "../util";
+import axios from "axios";
+import * as config from "../configuration";
 
-const recentAnalyses = `
-  SELECT j.id,
-         j.job_name as name,
-         j.job_description as description,
-         j.app_id,
-         j.app_name,
-         j.app_description,
-         j.result_folder_path,
-         j.start_date,
-         j.end_date,
-         j.planned_end_date ,
-         j.status,
-         j.subdomain,
-         j.parent_id,
-         u.username
-    FROM jobs j
-    JOIN users u ON j.user_id = u.id
-   WHERE j.deleted = false
-     AND u.username = $1
-ORDER BY start_date DESC
-   LIMIT $2
-`;
-
-export const getData = async (db, username, limit) => {
-    const { rows } = await db
-        .query(recentAnalyses, [username, limit])
-        .catch((e) => {
-            throw e;
-        });
-
-    if (!rows) {
-        throw new Error("no rows returned");
+export const getData = async (username, limit) => {
+    try {
+        const { data } = await axios.get(
+            `${config.appsURL}/analyses?sort-field=startdate&limit=${limit}&sort-dir=DESC&user=${username}`
+        );
+        logger.info(
+            "Recent analyses for user " + username + ": " + JSON.stringify(data)
+        );
+        return data;
+    } catch (e) {
+        throw new Error(e);
     }
-
-    return rows;
 };
 
-const getHandler = (db) => async (req, res) => {
+const getHandler = () => async (req, res) => {
     try {
         const limit = validateLimit(req?.query?.limit) ?? 10;
         const username = req?.params?.username;
-        const rows = await getData(db, username, limit);
-        res.status(200).json({ analyses: rows });
+        const rows = await getData(username, limit);
+        res.status(200).json(rows);
```

Comment:
This line and change is why I was wondering about data changes in the other endpoints. Presumably before we were getting an array of analyses and thus adding in the `{analyses: }` wrapper. Now, I imagine, we're getting an `{analyses: [...]}` directly from the endpoint. Since we also call the `getData` in this namespace in the `"/users/:username"` handler, that code was probably also assuming it was getting an array, not an object. As such, this:

```javascript
            analyses: {
                recent: await recent,
                running: await running,
            },
```
was probably originally producing something like:

```javascript
{
    analyses: {
        recent: [...],
        running: [...]
},
```

but is now probably looking like:

```javascript
{
    analyses: {
        recent: {
            analyses: [...]
        },
        running: {
            analyses: [...]
        },
},
```

I'm guessing the extra `{analyses: }` wrappers in there aren't intended and might be a bit cluttered in the UI code (`analyses.recent.analyses` rather than just `analyses.recent` all over the place).

Reply:
Ahh I see. Thanks for the comment.

Reply:
I was anticipating to make some changes in UI. So I had this comment earlier - https://github.com/cyverse-de/dashboard-aggregator/pull/9#issuecomment-928205282.